### DISCUSSION TITLE
Problemas de compilación en Angular 15

### DISCUSSION THREAD

**Discussion by lfernandezlong** [gcba/Obelisco#139] [Configuración]
Estoy usando Angular 15 en mi proyecto y dejaba de compilar al traer Obelisco. Averiguando me di cuenta que a partir de Angular 13 quedó deprecado el uso de ~ en los imports, y en cada .scss de Obelisco hay un import con ~ que llama a la hoja de ese componente en Bootstrap. 
Por favor, podrían cambiar los imports para que no esté deprecado o ofrecer alguna solución para poder usar versiones actuales de Angular?

**Comment by alan199912** [comment]
Gracias por ponerse en contacto con nosotros. Entendemos que ha estado experimentando problemas con la compilación de su proyecto Angular 15 al incorporar Obelisco, y que ha identificado un problema en el uso de los imports "~" en los archivos SCSS de Obelisco.

Nos gustaría entender mejor su situación y ayudarlo a resolver el problema. ¿Podría proporcionarnos más información acerca de su proyecto?

Necesitaríamos más detalles sobre cómo está utilizando Obelisco en su proyecto y qué tipo de errores está experimentando durante la compilación. Si pudiera proporcionarnos más detalles o capturas de pantalla de los errores, podríamos analizarlos y ofrecer soluciones más precisas.

También sería útil saber qué versión de Obelisco está utilizando y si ha actualizado recientemente la librería.

Agradeceríamos cualquier información adicional que pueda proporcionar para ayudarnos a entender mejor su situación y ofrecer soluciones precisas para resolver los problemas que está experimentando.

Si necesita más ayuda o tiene alguna pregunta adicional, no dude en contactarnos nuevamente. Estamos aquí para ayudar.

Quedamos a la espera de su respuesta.

**Reply by lfernandezlong** [reply]
Hola, buenas tardes.

Por pedido de cliente de GCBA arrancamos un proyecto en Angular este mes, nos comentaron que debía tener cualquier versión mayor a la 4, por ende utilizamos la última disponible que es la 15.2.
Tomando la misma estrategia que se usa en el resto de los proyectos para editar los estilos, importamos en un styles.scss propio el scss de Obelisco, con la línea "@import "@gcba/obelisco/src/scss/obelisco.scss";" que nos dio error de compilación: 
![image](https://user-images.githubusercontent.com/114437831/224746161-5abcef89-de1b-49ca-b849-0ac176d45fc6.png)

Averiguando me encontré con la solución de eliminar los ~ en imports de esta versión de Angular ([post de stackoverflow](https://stackoverflow.com/questions/74611743/sasserror-cant-find-stylesheet-to-import-angular-upgrade)), la primer idea fue importar los archivos de bootstrap y obelisco en un archivo propio para evitar este glifo en la importación, pero me encontré con el mismo problema porque en muchos .scss de Obelisco se utiliza esa forma de importación como por ejemplo se ve en la hoja de estilos de los alerts

![image](https://user-images.githubusercontent.com/114437831/224746094-9ec5cbb4-d635-474f-a73b-91cdd76ca22e.png)

Espero que se logre una solución a la brevedad, quedo a tu disposición.

Saludos

**Reply by alan199912** [reply]
Gracias por su respuesta y por compartir más información sobre su proyecto. Comprendo que ha estado utilizando la última versión disponible de Angular, que es la 15.2, y que ha estado importando los estilos de Obelisco en un archivo **styles.scss** propio utilizando la línea `@import "@gcba/obelisco/src/scss/obelisco.scss";`.

Me alegra poder ayudarlo a resolver los problemas que está experimentando con Obelisco en su proyecto. Sin embargo, la forma en que ha estado importando los estilos de Obelisco en su proyecto a través de una línea "@import" en el archivo styles.scss es incorrecta. Para utilizar los estilos de una librería en un proyecto de Angular, es necesario declarar la ruta del archivo CSS correspondiente en la sección **styles** del archivo `angular.json`.

En el caso de Obelisco, se debe agregar la ruta `node_modules/@gcba/obelisco/dist/obelisco.css` en la sección **styles**  de `angular.json`. Como se muestra a continuacion:

```
"styles": [
  "node_modules/@gcba/obelisco/dist/obelisco.css",
  "src/styles.css"
],
```

Le sugiero que realice esta modificación y pruebe si su proyecto puede compilar correctamente. Si aún tiene problemas, no dude en contactarnos nuevamente para que podamos ayudarlo a encontrar una solución.

Espero que esto sea de ayuda para usted. Quedo a la espera de sus comentarios.

**Reply by lfernandezlong** [reply]
De esa forma compila, pero no puedo tomar los estilos desde mi css para crear nuevos estilos a partir de los existentes en Obelisco, por eso necesito lograr importar el .scss de Obelisco en mi hoja.

**Reply by alan199912** [reply]
Tenga en cuenta que para poder utilizar los estilos de Obelisco y crear nuevos a partir de este, es necesario tener instalado Bootstrap en su proyecto, ya que Obelisco depende de Bootstrap en la versión 4.5.0

Le recomiendo que intente instalar Bootstrap en su proyecto utilizando el siguiente comando:

```
npm install bootstrap@4.5.0
```

Si después de instalar Bootstrap el problema persiste, por favor comuníquese con nosotros nuevamente y con gusto lo ayudaremos a resolver el problema.

**Reply by lfernandezlong** [reply]
El problema va a persistir porque no tiene que ver con la librería Bootstrap, el problema viene de la forma de importación. 
Probé en un caso de traer manualmente los archivos de bootstrap sacando la ~ y compiló, pero dentro de los archivos de Obelisco hay importaciones con "~" que hacen que el proyecto no compile.

**Reply by alan199912** [reply]
Por favor, intente actualizar la librería Obelisco a la version 1.46.2 y pruebe nuevamente su proyecto.
Si el problema persiste, por favor háganoslo saber y estaremos encantados de ayudarlo a resolverlo.

**Reply by alan199912** [reply]
Quería informarte que realizamos una actualización en la librería Obelisco para resolver el problema que estabas experimentando con la versión de Angular 15 y las importaciones con `~`.

La versión actualizada de Obelisco es la 1.46.2.

Te invitamos a probar nuevamente tu proyecto con esta nueva versión de la librería y nos dejes saber si el problema ha sido resuelto.

Por favor, ten en cuenta que si no recibimos una respuesta en el dia de hoy, consideraremos que el problema ha sido resuelto y daremos por cerrado este tema.

Si necesitas ayuda adicional, por favor no dudes en contactarnos y estaremos encantados de ayudarte.

Queremos agradecerte por utilizar Obelisco y esperamos poder seguir brindándote un servicio de calidad.
